chamber/utils/__init__.py

```diff
@@ -43,15 +47,19 @@ def keep_spacing(value, autoescape=True):
     return mark_safe(value.replace('  ', ' &nbsp;').replace('\n', '<br />'))
 
 
-def call_method_with_unknown_input(method, **fun_kwargs):
-    method_kwargs_names = inspect.getargspec(method)[0][1:]
-
-    method_kwargs = {arg_name: fun_kwargs[arg_name] for arg_name in method_kwargs_names if arg_name in fun_kwargs}
-
-    if len(method_kwargs_names) == len(method_kwargs):
-        return method(**method_kwargs)
+def call_function_with_unknown_input(function, **kwargs):
+    """
+    Call function and use kwargs from input if function requires them.
+    :param function: function to call.
+    :param kwargs: function input kwargs or extra kwargs which will not be used.
+    :return: function result or raised InvalidFunctionArguments exception.
+    """
+    function_kwargs_names = set(inspect.signature(function).parameters.keys())
```

Comment:
Shouldn't you differentiate between kwargs with default value and kwargs without it? (kwargs with default value don't have to be specified in input)

Reply:
Interesting idea. I will try it

Reply:
done